data/data/azure/vnet/public-lb.tf

```diff
@@ -5,9 +5,9 @@ locals {
 
 locals {
   // DEBUG: Azure apparently requires dual stack LB for v6
-  need_public_ipv4 = ! var.azure_private || var.azure_outbound_routing_type != "UserDefinedRouting"
+  need_public_ipv4 = ! var.azure_private || ! var.azure_lb_private || var.azure_outbound_routing_type != "UserDefinedRouting"
```

Comment:
Do we need a second `azure_lb_private` variable in addition to `azure_private`? It looks like it is the same as `azure_private` which would make sense to me: considering that Terraform only creates the API load balancers, I'm not sure we would need any changes to the terraform configs at all. I think we would only need the distinction if Terraform creates any resources related to ingress (or the registry if we're including that component) such as security groups.

Reply:
I was having this issue where terraform wasn't allowing me to associate the bootstrap NIC that has a public IP with a LB that has an outbound rule. I was trying to create a new NIC card for bootstrap only when the lb is private so I needed this variable. I think I need this for other reasons like converting the cname record to A-record on the DNS if lb is private.

Reply:
> I was having this issue where terraform wasn't allowing me to associate the bootstrap NIC that has a public IP with a LB that has an outbound rule.

Doesn't this already work when the publishing strategy is set to Internal or private?

Unless there are connections/overlap in the terraform configs with ingress/registry I don't see why we would need changes to Terraform, which should really simplify things.

Reply:
It works in internal if the bootstrap is also private. If the bootstrap is public and the lb is private we get into some trouble.

Reply:
> If the bootstrap is public and the lb is private we get into some trouble.


I don't think that is a valid config. If the cluster is private/internal bootstrap should not have a public IP.

https://github.com/openshift/installer/blob/9ff8208c93feb0569b77bf583c85087b8d638c44/data/data/azure/bootstrap/main.tf#L79

Reply:
But the requirement is to keep just the apiserver private and the others public.

Reply:
The bootstrap node's not a separate component like ingress or the registry--visibility should be the same as the apiserver in all cases. If there is indeed a specific requirement to have bootstrap visibility configurable as separate from the API server LMK.